internal/controller/controllers/cluster_controller.go

```diff
@@ -63,26 +64,32 @@ type ClusterReconciler struct {
 func (r *ClusterReconciler) Reconcile(req ctrl.Request) (ctrl.Result, error) {
 	ctx := context.Background()
 	cluster := &adiiov1alpha1.Cluster{}
+	foundCRD := true
```

Comment:
why do you need it? why not to put the logic under `k8serrors.IsNotFound(err)` if crd wasn't found try to delete from db and ignore 404

Reply:
you mean to use the deregister cluster API or to delete directly from DB?
if calling to deregister cluster api so we do need to get the cluster from DB, that is why I did it like that.

Reply:
I understand but this way it looks very confusing, so why not do it as part of a dedicated delete function?
Definitely not delete from the DB directly.